weld/llvm.rs

```diff
@@ -249,23 +267,359 @@ impl LlvmGenerator {
+            ctx.code.add(format!("{} = call i64 {}.size({} {})",
+                data_size_ll_tmp, data_prefix, data_ll_ty, data_ll_sym));
+
+            // Obtain the start and stride values.
+            let (start_str, stride_str) = if iter.start.is_none() {
+                // We already checked to make sure the FringeIter doesn't have a start, etc.
+                let start_str = match iter.kind {
+                    IterKind::FringeIter => fringe_start_str.as_ref().unwrap().to_string(),
+                    _ => String::from("0")
+                };
+                let stride_str = String::from("1");
+                (start_str, stride_str)
+            } else {
+                (llvm_symbol(iter.start.as_ref().unwrap()), llvm_symbol(iter.stride.as_ref().unwrap()))
+            };
+
+            let t0 = ctx.var_ids.next();
+            let t1 = ctx.var_ids.next();
+            let t2 = ctx.var_ids.next();
+            let cond = ctx.var_ids.next();
+            let next_bounds_check_label = ctx.var_ids.next();
+
+            // TODO just compare against end here...this computation is redundant.
+            // t0 = sub i64 num_iters, 1
+            // t1 = mul i64 stride, t0
+            // t2 = add i64 t1, start
+            // cond = icmp lte i64 t1, size
+            // br i1 cond, label %nextCheck, label %checkFailed
+            // nextCheck:
+            // (loop)
+            ctx.code.add(format!("{} = sub i64 {}, 1", t0, num_iters_str));
+            ctx.code.add(format!("{} = mul i64 {}, {}", t1, stride_str, t0));
+            ctx.code.add(format!("{} = add i64 {}, {}", t2, t1, start_str));
+            ctx.code.add(format!("{} = icmp ult i64 {}, {}", cond, t2, data_size_ll_tmp));
+            ctx
+                .code
+                .add(format!("br i1 {}, label {}, label %fn.boundcheckfailed", cond, next_bounds_check_label));
+            ctx.code.add(format!("{}:", next_bounds_check_label.replace("%", "")));
+        }
+
+        // If we get here, the bounds check passed.
+        ctx.code.add(format!("br label %fn.boundcheckpassed"));
+        // Handle a bounds check fail.
+        ctx.code.add(format!("fn.boundcheckfailed:"));
+        let errno = WeldRuntimeErrno::BadIteratorLength;
+        let run_id = ctx.var_ids.next();
+        ctx.code.add(format!("{} = call i64 @weld_rt_get_run_id()", run_id));
+        ctx.code.add(format!("call void @weld_run_set_errno(i64 {}, i64 {})", run_id, errno as i64));
+        ctx.code.add(format!("call void @weld_rt_abort_thread()"));
+        ctx.code.add(format!("; Unreachable!"));
+        ctx.code.add(format!("br label %fn.end"));
+        ctx.code.add(format!("fn.boundcheckpassed:"));
+
+        Ok(())
+    }
+
+    /// Generates code to either call into the parallel runtime or to call the function which
+    /// executes the for loops body directly on a single thread, given the number of iterations to
+    /// be executed.
+    /// 
+    /// Succeeds gen_loop_bounds_check
```

Comment:
By `Succeeds`, do you mean `Follows`? I would change the wording there to make it clearer.